src/lightning/fabric/utilities/data.py

```diff
@@ -196,7 +197,9 @@ def _dataloader_init_kwargs_resolve_sampler(
                 )
 
             batch_sampler = _reinstantiate_wrapped_cls(batch_sampler, *args, **kwargs)
-        else:
+        elif hasattr(batch_sampler, "batch_size") and hasattr(batch_sampler, "drop_last"):
```

Comment:
Having these attributes doesn't guarantee that they will be available on `__init__`. You would have to inspect the signature for this

Reply:
Yes that's right, however the approach so far prior to the PR was to just try it and handle the exception if one occurs. This still required these two attributes, but if they aren't present then it is not even possible to try re-instantiating the sampler.